ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java

```diff
@@ -0,0 +1,174 @@
+      properties = new HashMap<>();
+      properties.putAll(abstractStreamCreateStatement.getProperties());
+      properties.put(KsqlConstants.AVRO_SCHEMA_ID, schemaIdLiteral);
+    }
+
+
+    if (abstractStreamCreateStatement instanceof CreateStream) {
+      return new CreateStream(abstractStreamCreateStatement.getName(), elements,
+                              ((CreateStream) abstractStreamCreateStatement).isNotExists(),
+                              properties);
+    } else if (abstractStreamCreateStatement instanceof CreateTable) {
+      return new CreateTable(abstractStreamCreateStatement.getName(), elements,
+                             ((CreateStream) abstractStreamCreateStatement).isNotExists(),
+                             properties);
+    } else {
+      throw new KsqlException(String.format("Invalid AbstractStreamCreateStatement: %s",
+                                            SqlFormatter.formatSql(abstractStreamCreateStatement)));
+    }
+  }
+
+
+  public static synchronized void validatePersistantQueryResults(PersistentQueryMetadata
+                                                                     persistentQueryMetadata,
+                                                                 SchemaRegistryClient
+                                                                     schemaRegistryClient) {
+    if (persistentQueryMetadata.getResultTopic().getKsqlTopicSerDe().getSerDe() == DataSource.DataSourceSerDe.AVRO) {
+      String avroSchemaString = new MetastoreUtil().buildAvroSchema(persistentQueryMetadata
+                                                                        .getResultSchema(),
+                                                                    persistentQueryMetadata
+                                                                        .getResultTopic().getName());
+      boolean isValidSchema = AvroUtil.isValidAvroSchemaForTopic(persistentQueryMetadata.getResultTopic()
+                                                                     .getTopicName(),
+                                                                 avroSchemaString, schemaRegistryClient);
+      if (!isValidSchema) {
+        throw new KsqlException(String.format("Cannot register avro schema for %s since it is "
+                                              + "not valid for schema registry.", persistentQueryMetadata
+                                                  .getResultTopic().getKafkaTopicName()));
+      }
+    }
+  }
+
+
+  private static synchronized boolean isValidAvroSchemaForTopic(String topicName, String
+      avroSchemaString, SchemaRegistryClient schemaRegistryClient) {
+
+    org.apache.avro.Schema.Parser parser = new org.apache.avro.Schema.Parser();
+    org.apache.avro.Schema avroSchema = parser.parse(avroSchemaString);
+    try {
+      return schemaRegistryClient.testCompatibility(topicName, avroSchema);
+    } catch (IOException e) {
+      log.error(ExceptionUtil.stackTraceToString(e));
```

Comment:
perhaps some information in the logs? i.e, the topic and schema etc

Reply:
Done!